src/Lean/MetavarContext.lean

```diff
@@ -264,6 +268,8 @@ structure MetavarDecl where
   `mvarIdPending` is a `syntheticOpaque` metavariable that has not been synthesized yet. The delayed assignment becomes a real one
   as soon as `mvarIdPending` has been fully synthesized.
   `fvars` are variables in the `mvarIdPending` local context.
+
+  See `instantiateLevelMVars` for the rationale of delayed assignments.
```

Comment:
```suggestion
  See the comment below `assignDelayedMVar ` for the rationale of delayed assignments.
```
I think it would make sense to inline that comment here.

Reply:
@Kha I am not so confident, because I would then want to add a comment at `assignDelayedMVar` saying:

> see the comment above `DelayedMetavarAssignment` for more details.

Perhaps we should adopt the [`[Note]` system of GHC](https://github.com/ghc/ghc/search?p=3&q=Note+%5B+%5D&type=) as a lightweight mechanism to reference these kinds of details?

Reply:
FYI, mathlib already uses that system, see https://leanprover-community.github.io/mathlib_docs/commands.html#library_note . If desired we could move the `library_note` command to core.

Reply:
Yeah, `library_note` seems nice. I think a short RFC in that regard would be great.